### PR TITLE
make mysqli_result generic

### PR DESCRIPTION
**Pull request by staabm**
analog https://github.com/phpstan/phpstan-src/pull/890

inspired by psalm: https://github.com/vimeo/psalm/blob/10167a223a073c6ce3fb1064c46082afad6993cc/stubs/mysqli.phpstub

### CODE REVIEW

File: conf/config.neon [hunk 1]
```diff
@@ -22,7 +22,9 @@ parameters:
 	featureToggles:
 		bleedingEdge: false
 		disableRuntimeReflectionProvider: false
-		skipCheckGenericClasses: []
+		skipCheckGenericClasses: [
+			'mysqli'
```
Comment (ondrejmirtes) [inline conf/config.neon:26]: You're making `mysqli_result` generic, not `mysqli`

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: stubs/mysqli.stub [hunk 2]
```diff
@@ -49,3 +49,31 @@ class mysqli_stmt
...
+     * @param class-string<T> $class
+     * @param array<mixed> $constructor_args
+     * @return T|null|false
+     */
+    function fetch_object(string $class = stdClass::class, array $constructor_args = []): object|false|null {}
+}
+
+/**
+ * @template T of object
+ * @template TValue
```
Comment (staabm) [inline stubs/mysqli.stub:72]:
```suggestion
 * @template TValue of object
```

**Comment by ondrejmirtes** [comment]
Can you please also test various `mysqli_fetch_object` calls? The situation is quite complex so I'd like to verify the generic function works. Thank you.

**Comment by staabm** [comment]
added a few positive tests. do you have something more in mind, when you say "various" ? :-)

**Comment by staabm** [comment]
since I have put these stubs in phpstan-dba, I will close this PR.

having them inside the database related repo gives me more flexibility and I think is the best thing long term, as phpstan-src needs to stay more generic and cannot per default be as opinionated as phpstan-dba might needs to be

**Pull request closed without merging**